app/src/views/Game.vue

```diff
@@ -112,27 +116,39 @@ export default {
       }
       return this.gameState.deck.length;
     },
+    gamePhase() {
+      return this.gameState?.specialPhase;
+    },
     isPerishPhase() {
       return this.gameState?.specialPhase === GamePhase.ResolvingPerish;
     },
     isUserDead() {
       return this.gameState?.statuses[this.userId] == PlayerStatus.Dead;
+    },
+    playerCount() {
+      return this.gameState?.players.length;
     }
   },
 
   watch: {
     isPerishPhase(value, oldValue) {
+      if(value !== GamePhase.ResolvingPerish) {
+        clearInterval(counter);
+      }
+
       if (this.isUserTurn && value && !oldValue) {
-        this.resolveCountdown = 3;
-        const cb = () => {
-          this.resolveCountdown--;
-          if (this.resolveCountdown > 0) {
-            setTimeout(cb, 1000);
-          } else {
-            this.solvePerish();
+        this.resolveCountdown = 1000;
+        
+        const timer = () => {
+          if (this.resolveCountdown <= 0) {
+            this.perish();
+            clearInterval(counter);
+            return;
           }
+          this.resolveCountdown--;
         };
-        setTimeout(cb, 1000);
+
+        var counter = setInterval(timer, 10);
```

Comment:
IMO, it would be more "legible" to declare the `counter` variable at the beginning of the function than to use some `var` time traveling shenanigans

```javascript
isPerishPhase(value, oldValue) {
  let counter = null;
  if(value !== GamePhase.ResolvingPerish) {
    clearInterval(counter);
  }
```